public/app/plugins/datasource/cloudwatch/config_ctrl.ts

```diff
@@ -36,4 +40,42 @@ export class CloudWatchConfigCtrl {
     { name: 'Monthly', value: 'Monthly', example: '[logstash-]YYYY.MM' },
     { name: 'Yearly', value: 'Yearly', example: '[logstash-]YYYY' },
   ];
+
+  regions = [
+    'ap-northeast-1',
+    'ap-northeast-2',
+    'ap-northeast-3',
+    'ap-south-1',
+    'ap-southeast-1',
+    'ap-southeast-2',
+    'ca-central-1',
+    'cn-north-1',
+    'cn-northwest-1',
+    'eu-central-1',
+    'eu-north-1',
+    'eu-west-1',
+    'eu-west-2',
+    'eu-west-3',
+    'me-south-1',
+    'sa-east-1',
+    'us-east-1',
+    'us-east-2',
+    'us-gov-east-1',
+    'us-gov-west-1',
+    'us-iso-east-1',
+    'us-isob-east-1',
+    'us-west-1',
+    'us-west-2',
+  ];
+
+  getRegions() {
+    this.datasourceSrv
+      .loadDatasource(this.current.name)
+      .then(ds => {
+        return ds.getRegions();
+      })
+      .then(regions => {
+        this.regions = _.uniq(this.regions.concat(_.map(regions, 'value')));
```

Comment:
It should work in isolated region without merging.

Grafana only use hardcoded region list in frontend code, when fail to get region list from backend.

If success to get region from backend, it always add isolated regions to the response.
https://github.com/grafana/grafana/pull/12337/files/553a51f8ac8e7a3e8c0f53c14a10cbfb884d51b6#diff-1bc1bcaa896482730fda8a2d539fb247R237
https://github.com/grafana/grafana/pull/12337/files/553a51f8ac8e7a3e8c0f53c14a10cbfb884d51b6#diff-1bc1bcaa896482730fda8a2d539fb247R249

The behavior is little tricky, because if Grafana version is old, and there are new region not in hardcoded region list, the region list is updated after datasource created.
But this behavior will help the user who want to use new region.

Reply:
> Grafana only use hardcoded region list in frontend code, when fail to get region list from backend.

This is not true as I understand the code. `this.regions = _.uniq(this.regions.concat(_.map(regions, 'value')));` will merge hardcoded list with backend response if I'm not mistaken. As long as this behavior is intentional it's fine. We can look into the "only showing of regions that I can access" to later. Is the behavior intentional? If so, we can merge this.

> If success to get region from backend, it always add isolated regions to the response.

Yes that's good solution for isolated regions.

> But this behavior will help the user who want to use new region.

Agree, good point

Reply:
Sorry, I missed the point, and understand what your intention now.

If backend response is success, merge the response to hardcoded list. But it is not required, because backend response should include all valid region list. So I change the code, just copy backend response.

And add error message for who can check Dev tools.